File: 백준/Silver/10866. 덱/덱.py
import sys
from collections import deque

N = int(sys.stdin.readline())
d = deque()

for _ in range(N):
    command = sys.stdin.readline().split()

    if command[0] == "push_front":
        d.appendleft(int(command[1]))

    elif command[0] == "push_back":
        d.append(int(command[1]))

    elif command[0] == "pop_front":
        if not d:
            print(-1)
        else:
            print(d.popleft())

    elif command[0] == "pop_back":
        if not d:
            print(-1)
        else:
            print(d.pop())

    elif command[0] == "size":
        print(len(d))

    elif command[0] == "empty":
        if not d:
            print(1)
        else: print(0)

    elif command[0] == "front":
        if not d:
            print(-1)
        else:
            print(d[0])

    else:
        if not d:
            print(-1)
        else:
            print(d[-1])

File: 백준/Silver/10866. 덱/덱.java
import java.util.*;
import java.io.*;

class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));

        int N = Integer.parseInt(br.readLine());

        LinkedList<Integer> deque = new LinkedList<>();

        while (N-- > 0) {
            StringTokenizer st = new StringTokenizer(br.readLine());
            String cmd = st.nextToken();
            switch (cmd) {
                case "push_front":
                    deque.addFirst(Integer.parseInt(st.nextToken()));
                    break;

                case "push_back":
                    deque.addLast(Integer.parseInt(st.nextToken()));
                    break;

                case "pop_front":
                    if (deque.isEmpty()) {
                        bw.write("-1\n");
                    } else {
                        bw.write(deque.removeFirst() + "\n");
                    }
                    break;

                case "pop_back":
                    if (deque.isEmpty()) {
                        bw.write("-1\n");
                    } else {
                        bw.write(deque.removeLast() + "\n");
                    }
                    break;

                case "size":
                    bw.write(deque.size() + "\n");
                    break;

                case "empty":
                    if (deque.isEmpty()) {
                        bw.write("1\n");
                    } else {
                        bw.write("0\n");
                    }
                    break;

                case "front":
                    if (deque.isEmpty()) {
                        bw.write("-1\n");
                    } else {
                        bw.write(deque.getFirst() + "\n");
                    }
                    break;

                case "back":
                    if (deque.isEmpty()) {
                        bw.write("-1\n");
                    } else {
                        bw.write(deque.getLast() + "\n");
                    }
                    break;
            }
        }
        bw.flush();
        br.close();
        bw.close();
    }
}